### PR TITLE
Data source fix

### PR DESCRIPTION
**Pull request by FlanFlanagan**
A key error catch for missing cross sections within a cross section cache when building TAPE9.INP files. Probably needs some refinement.

### CODE REVIEW

File: pyne/origen22.py [hunk 1]
```diff
@@ -1509,18 +1507,22 @@ def xslibs(nucs=NUCS, xscache=None, nlb=(201, 202, 203), verbose=False):
...
-        if verbose:
-            print('computing {0}'.format(nucname.name(nuc)))
-        key = nucname.zzaaam(nuc)
-        if nuc in ACTIVATION_PRODUCT_NUCS:
-            _compute_xslib(nuc, key, t9[nlb[0]], xscache)
-        if nuc in ACTINIDE_AND_DAUGHTER_NUCS:
-            _compute_xslib(nuc, key, t9[nlb[1]], xscache)
-        if nuc in FISSION_PRODUCT_NUCS:
-            _compute_xslib(nuc, key, t9[nlb[2]], xscache)
+        try:
```
Comment (scopatz) [inline pyne/origen22.py]: Try blocks should be a couple of lines at most, and ideally should only be 1 line This is too all-encompassing.  I think you should have three try-except blocks, one in each of the conditionals.

File: pyne/xs/cache.py [hunk 2]
```diff
@@ -126,11 +128,11 @@ def __setitem__(self, key, value):
             value = _valid_group_struct(value)
             cache_value = self._cache['E_g']
             if _same_arr_or_none(value, cache_value):
-                return 
+                return
             self.clear()
             self._cache['phi_g'] = None
             for ds in self.data_sources:
-                ds.dst_group_struct = value
+               ds.dst_group_struct = value
```
Comment (scopatz) [inline pyne/xs/cache.py]: Why is this a 3-space indent? Shouldn't it be 4-space?

**Comment by scopatz** [comment]
Thanks for putting this in @FlanFlanagan!

**Comment by scopatz** [comment]
Also, could you please add a test to ensure that the new KeyError issue happens?  This should be pretty easy to do since you can create a cache with no data sources and then assert that it raises a KeyError whenever you try to get anything out of it.

**Comment by FlanFlanagan** [comment]
@scopatz ping. I fixed the test, I think.

**Comment by scopatz** [comment]
Thanks @FlanFlanagan!

**Pull request merged**